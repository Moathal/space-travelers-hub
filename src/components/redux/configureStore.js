import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions';
import rocketsReducer from './rockets';

export default configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice,
  },
});
