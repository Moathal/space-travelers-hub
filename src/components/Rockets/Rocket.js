import React from 'react';
/* import { useDispatch } from 'react-redux';
import { reserve } from '../redux/rockets'; */
import {
  RocketCard,
  ReserveButton,
  IMG,
  DetailsBlock,
  Badge,
} from './Rockets.styled';

function Rocket(/* {
  img,
  name,
  description,
  id,
  reserved,
  reservation,
} */) {
  /*  const dispatch = useDispatch();
  const handleClick = () => {
    dispatch(reserve(id));
  }; */
  return (
    <RocketCard>
      <IMG alt="Rocket Image" />
      <DetailsBlock>
        <h2>{/* {name} */}</h2>
        <p>
          <Badge>Reserved</Badge>
          {/* {description} */}
        </p>
        <ReserveButton type="button">
          {/* {reservation} */}
        </ReserveButton>
      </DetailsBlock>
    </RocketCard>
  );
}

export default Rocket;
