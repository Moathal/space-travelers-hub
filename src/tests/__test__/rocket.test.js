import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../components/redux/configureStore';
import Rocket from '../../components/Rockets/Rocket';

it('renders Rocket correctly', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <Rocket />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
