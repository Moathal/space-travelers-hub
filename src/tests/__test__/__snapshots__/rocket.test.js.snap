// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Rocket correctly 1`] = `
<div
  className="sc-bcXHqe ibjEgP"
>
  <img
    alt="Rocket Image"
    className="sc-hLBbgP vVmau"
  />
  <div
    className="sc-gswNZR jBXjgz"
  >
    <h2 />
    <p>
      <small
        className="sc-eDvSVe fSYAQR"
      >
        Reserved
      </small>
    </p>
    <button
      className="sc-dkrFOg iLVZrd"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
